pulpcore/cli/common/generic.py

```diff
@@ -183,30 +171,46 @@ def _version_callback(
     return value
 
 
-def load_json_callback(
+# TODO: would be great to have enable this to take a validator, rather than having
+# to build "on top of" it like I'm doing now w/ json_callback
+def load_file_or_string_callback(
     ctx: click.Context,
     param: click.Parameter,
     value: Optional[str],
 ) -> Any:
-    """Load JSON from input string or from file if string starts with @."""
-    json_object: Any
-    json_string: Union[str, bytes]
+    """Load string from input or from file if string starts with @."""
+    the_content: str
 
     # pass None and "" verbatim
     if not value:
         return value
 
     if value.startswith("@"):
-        json_file = value[1:]
+        the_file = value[1:]
         try:
-            with click.open_file(json_file, "rb") as fp:
-                json_string = fp.read()
+            with click.open_file(the_file, "r") as fp:
+                the_content = fp.read()
         except OSError:
             raise click.ClickException(
-                _("Failed to load content from {json_file}").format(json_file=json_file)
+                _("Failed to load content from {json_file}").format(json_file=the_file)
             )
     else:
-        json_string = value
+        the_content = value
+
+    return the_content
+
+
+def load_json_callback(
+    ctx: click.Context,
+    param: click.Parameter,
+    value: Optional[str],
+) -> Any:
+    """Load JSON from input string or from file if string starts with @."""
+    json_object: Any
+    json_string: str
+    json_string = load_file_or_string_callback(ctx, param, value)
+    if not json_string:
+        return value
```

Comment:
`""` would evaluate as proper json, so only `None` is really interesting and we can catch that before calling the other hook.